span.go

```diff
@@ -253,6 +253,9 @@ type Span struct {
 	transactionID SpanID
 	parentID      SpanID
 	exit          bool
+	ctxPropagated bool
+	composite     compositeSpan
+	buffer        spanBuffer
```

Comment:
IIANM these are only used in Span.End -- is that correct? If so, I think we should move them to SpanData. SpanData is discarded when the span is ended and its memory is recycled.